scripts/globals/bcnm.lua

```diff
@@ -1,761 +1,956 @@
+        [1024] = function() return ( cop > WHEN_ANGELS_FALL or (cop == WHEN_ANGELS_FALL and copStat > 4)                                            ) end, -- PM8-3: When Angels Fall
+        [1056] = function() return ( cop > DAWN or (cop == DAWN and copStat > 2)                                                                    ) end, -- PM8-4: Dawn
+    };
+
+    -- determine whether player meets cutscene skip requirements
+    local req = skipReqs[bfid];
+    if (req == nil) then
+        return false;
+    elseif (req()) then
+        return true;
+    else
+        return false;
     end
 end
 
-function EventTriggerBCNM(player, npc)
-    player:setVar("trade_bcnmid", 0)
-    player:setVar("trade_itemid", 0)
+-----------------------------------------------
+-- which battlefields are valid for registrant?
+-----------------------------------------------
 
-    if (player:hasStatusEffect(EFFECT_BATTLEFIELD)) then
-        if (player:isInBcnm() == 1) then
-            player:startEvent(0x7d03) -- Run Away or Stay menu
-        else -- You're not in the BCNM but you have the Battlefield effect. Think: non-trader in a party
-            status = player:getStatusEffect(EFFECT_BATTLEFIELD)
-            playerbcnmid = status:getPower()
-            playermask = GetBattleBitmask(playerbcnmid, player:getZoneID(), 1)
-            if (playermask~=-1) then
-                -- This gives players who did not trade to go in the option of entering the fight
-                player:startEvent(0x7d00, 0, 0, 0, playermask, 0, 0, 0, 0)
-            else
-                player:messageBasic(94, 0, 0)
-            end
-        end
-        return true
-    else
-        if (checkNonTradeBCNM(player, npc)) then
-            return true
+function findBattlefields(player, npc, itemId)
+    local mask = 0;
+    local zbfs = battlefields[player:getZoneID()];
+    if (zbfs == nil) then
+        return 0;
+    end
+    for k, v in pairs(zbfs) do
+        if (v[3] == itemId and checkReqs(player, npc, v[2], true)) then
+            mask = bit.bor(mask,math.pow(2,v[1]));
         end
     end
-
-    return false
+    return mask;
 end
 
-function EventUpdateBCNM(player, csid, option, entrance)
-    -- return false
-    local id = player:getVar("trade_bcnmid") -- this is 0 if the bcnm isnt handled by new functions
-    local skip = CutsceneSkip(player, npc)
+-----------------------------------------------
+-- get battlefield id for a given zone and bit
+-----------------------------------------------
 
-    print("UPDATE csid "..csid.." option "..option)
-    -- seen: option 2, 3, 0 in that order
-    if (csid == 0x7d03 and option == 2) then -- leaving a BCNM the player is currently in.
-        player:updateEvent(3)
-        return true
-    elseif (csid == 0x7d03 and option == 3) then -- leaving a BCNM the player is currently in.
-        player:u
```

Comment:
you'll wanna handle party members registration too with return codes, and depending on that in 0x05C handler update their position
https://github.com/DarkstarProject/darkstar/blob/bcnmshit/scripts/globals/bcnm.lua#L503